homeassistant/components/zwave_js/config_flow.py

```diff
@@ -726,13 +737,109 @@ def _async_update_entry(self, data: dict[str, Any]) -> None:
 
         return await self.async_step_manual()
 
+    async def async_step_intent_migrate(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm the user wants to reset their current radio."""
+        if not self.config_entry.data.get(CONF_USE_ADDON):
+            return self.async_abort(reason="addon_required")
+
+        if user_input is not None:
+            return await self.async_step_backup_nvm()
+
+        return self.async_show_form(step_id="intent_migrate")
+
+    async def async_step_backup_nvm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Backup the current network."""
+        if self.backup_task is None:
+            self.backup_task = self.hass.async_create_task(self._async_backup_network())
+
+        if not self.backup_task.done():
+            return self.async_show_progress(
+                step_id="backup_nvm",
+                progress_action="backup_nvm",
+                progress_task=self.backup_task,
+            )
+
+        try:
+            await self.backup_task
+        except AbortFlow as err:
+            _LOGGER.error(err)
+            return self.async_show_progress_done(next_step_id="backup_failed")
+        finally:
+            self.backup_task = None
+
+        return self.async_show_progress_done(next_step_id="instruct_unplug")
+
+    async def async_step_restore_nvm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Restore the backup."""
+        if self.restore_backup_task is None:
+            self.restore_backup_task = self.hass.async_create_task(
+                self._async_restore_network_backup()
+            )
+
+        if not self.restore_backup_task.done():
+            return self.async_show_progress(
+                step_id="restore_nvm",
+                progress_action="restore_nvm",
+                progress_task=self.restore_backup_task,
+            )
+
+        try:
+            await self.restore_backup_task
+        except AbortFlow as err:
+            _LOGGER.error(err)
+            return self.async_show_progress_done(next_step_id="restore_failed")
+        finally:
+            self.restore_backup_task = None
+
+        return self.async_show_progress_done(next_step_id="migration_done")
+
+    async def async_step_instruct_unplug(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Reset the current radio, and instruct the user to unplug it."""
+
+        if user_input is not None:
+            # Now that the old radio is gone, we can scan for serial ports again
+            return await self.async_step_choose_serial_port()
+
+        client: Client = self.config_entry.runtime_data[DATA_CLIENT]
+        if client.driver is None:
+            return self.async_abort(reason="driver_not_ready")
+        # reset the old radio
+        await client.driver.async_hard_reset()
```

Comment:
This currently doesn't destroy the cached client driver state. This may not matter during the migration, but if the user would stop the migration after this step, the current state of the integration is incorrect since the stored client driver state is stale.

I don't think we need to address this in this PR, but we should keep it in mind and follow up. I've been planning to improve the hard reset feature, but haven't come around to that yet.

Reply:
Shouldn't we have an additional confirmation before resetting the old controller? Every destructive action should have explicit approval from the user I think. I'd move this to a separate step and let the user decide.

Reply:
It's how ZHA does it too. Warn at the start, then get on with it

Reply:
But can't we make this optional?

Reply:
I guess it makes it a bit more bullet in your own foot proof, to make it required.

Reply:
Making it optional would complicate the flow more and I don't think it's worth it. Also if resetting is optional, then unplugging after would be required